plugin/handler_socket/handlersocket/database.cpp

```diff
@@ -552,6 +552,8 @@ dbcontext::modify_record(dbcallback_i& cb, TABLE *const table,
   if (mod_op == 'U') {
     /* update */
     handler *const hnd = table->file;
+    table->mark_columns_needed_for_update();
+    hnd->prepare_for_row_logging();
```

Comment:
> We always want `row` format when using the handlersocket, right? Even if the server is configured with `statement`?

yes. As mentioned constructing a statement seems overkill.

> I wonder if we should warn in such a case (to at least suggest configuring with `mixed` format),

I looked at the protocol and there's no extension for a warning - https://github.com/DeNA/HandlerSocket-Plugin-for-MySQL/blob/master/docs-en/protocol.en.txt

With MDEV-7389 however grabbing warnings and putting them in an audit log is possible. So worth while supporting.

> or perhaps try and construct a matching query to log in `statement` format (though that seems fickle, and more worth than it would be work).

It does a bit. There a case where event DML of tables is changed to statement without a warning so not totally without precedence to ignore the binlog format setting.